sdk/servicebus/Azure.Messaging.ServiceBus/src/Client/ServiceBusTransportMetrics.cs

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Messaging.ServiceBus
+{
+    /// <summary>
+    /// A set of metrics that can be used to monitor communication between the client and service.
+    /// </summary>
+    public class ServiceBusTransportMetrics
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ServiceBusTransportMetrics"/> class for mocking.
+        /// </summary>
+        protected internal ServiceBusTransportMetrics()
+        {
+        }
+        /// <summary>
+        /// Gets the last time that a heartbeat was received from the Service Bus service. These heartbeats are sent from the
+        /// service approximately every 30 seconds.
+        /// </summary>
+        public virtual DateTimeOffset? LastHeartBeat { get; internal set; }
```

Comment:
Do these need to be virtual?   Is there a scenario where creating an instance via the model factory wouldn't be sufficient?

Reply:
A model factory would work as well.